test/mason/publish-tests/publish.chpl

```diff
@@ -0,0 +1,18 @@
+
+use MasonPublish;
+use MasonUtils;
+
+proc main() throws {
+  try! {
+    const args = ['mason', 'publish'];
+    runCommand('mason new test_dir');
```

Comment:
Can we switch these to mason library calls rather than command line so that we don't require mason to be built to run it?